src/shared/ShutterTimer/useGetValidatorRegistryLogs.ts

```diff
@@ -0,0 +1,42 @@
+import { useQuery } from '@tanstack/react-query';
+import { providers, Contract } from 'ethers';
+
+import { CHAINS_MAP } from '@/constants/chains';
+import validatorRegistryABI from '@/abis/validatorRegistryABI';
+
+// query
+const LOGS_QUERY_KEY = 'logs';
+export const useGetValidatorRegistryLogs = (chainId: number) => useQuery({
+  queryKey: [LOGS_QUERY_KEY, chainId],
+  queryFn: async () => {
+    try {
+      const chain = CHAINS_MAP[chainId];
+      const rpc = chain.rpcUrls.default.http[0];
+      const provider = new providers.JsonRpcProvider(rpc);
+
+      const responseLogs = await provider.getLogs({
+        address: chain.contracts.validatorRegistry.address,
+        topics: [],
+        fromBlock: 'earliest',
```

Comment:
Could it be possible to cache the logs in a json file and save the latest block fetched, to avoid get the logs from genesis block at every query?